flytekit/annotated/reference_entity.py

```diff
@@ -0,0 +1,200 @@
+from typing import Any, Dict, Tuple, Type, Union
+
+from flytekit.annotated.context_manager import BranchEvalMode, ExecutionState, FlyteContext
+from flytekit.annotated.interface import Interface, transform_interface_to_typed_interface
+from flytekit.annotated.node import create_and_link_node
+from flytekit.annotated.promise import Promise, VoidPromise, create_task_output, translate_inputs_to_literals
+from flytekit.annotated.type_engine import TypeEngine
+from flytekit.common.exceptions import user as _user_exceptions
+from flytekit.loggers import logger
+from flytekit.models import dynamic_job as _dynamic_job
+from flytekit.models import interface as _interface_models
+from flytekit.models import literals as _literal_models
+from flytekit.models.core import identifier as _identifier_model
+
+
+class Reference(object):
+    def __init__(
+        self, resource_type: int, project: str, domain: str, name: str, version: str, *args, **kwargs,
+    ):
+        self._id = _identifier_model.Identifier(resource_type, project, domain, name, version)
+
+    @property
+    def id(self) -> _identifier_model.Identifier:
+        return self._id
+
+
+class TaskReference(Reference):
+    def __init__(
+        self, project: str, domain: str, name: str, version: str, *args, **kwargs,
+    ):
+        super().__init__(_identifier_model.ResourceType.TASK, project, domain, name, version, *args, **kwargs)
+
+
+class LaunchPlanReference(Reference):
+    def __init__(
+        self, project: str, domain: str, name: str, version: str, *args, **kwargs,
+    ):
+        super().__init__(_identifier_model.ResourceType.LAUNCH_PLAN, project, domain, name, version, *args, **kwargs)
+
+
+class WorkflowReference(Reference):
+    def __init__(
+        self, project: str, domain: str, name: str, version: str, *args, **kwargs,
+    ):
+        super().__init__(_identifier_model.ResourceType.WORKFLOW, project, domain, name, version, *args, **kwargs)
+
+
+class ReferenceEntity(object):
+    def __init__(
+        self,
+        resource_type,
```

Comment:
instead of Resource_type, project, domain etc, can we not just accept `Reference` type?

Reply:
I think so, let me do this.